precompiles/avs/types.go

```diff
@@ -14,52 +14,56 @@ func (p Precompile) GetAVSParamsFromInputs(_ sdk.Context, args []interface{}) (*
 
-	avsAddress, ok := args[1].(string)
+	avsName, ok := args[1].(string)
 	if !ok {
-		return nil, xerrors.Errorf(exocmn.ErrContractInputParaOrType, 1, "[]byte", avsAddress)
-	}
-	avsAddress, err := util.ProcessAvsAddress(avsAddress)
-	if err != nil {
-		return nil, xerrors.Errorf(exocmn.ErrContractInputParaOrType, 1, "[]byte", avsAddress)
+		return nil, xerrors.Errorf(exocmn.ErrContractInputParaOrType, 1, "string", avsName)
 	}
-	avsParams.AvsAddress = avsAddress
+	avsParams.AvsName = avsName
 
-	operatorAddress, ok := args[2].(string)
-	if !ok || operatorAddress == "" {
-		return nil, xerrors.Errorf(exocmn.ErrContractInputParaOrType, 2, "[]byte", operatorAddress)
+	slashContractAddr, ok := args[2].(string)
+	if !ok || slashContractAddr == "" {
+		return nil, xerrors.Errorf(exocmn.ErrContractInputParaOrType, 2, "string", slashContractAddr)
 	}
 
-	operatorAddress, err = util.ProcessAvsAddress(operatorAddress)
+	slashContractAddr, err = util.ProcessAvsAddress(slashContractAddr)
 	if err != nil {
-		return nil, xerrors.Errorf(exocmn.ErrContractInputParaOrType, 1, "[]byte", operatorAddress)
+		return nil, xerrors.Errorf(exocmn.ErrContractInputParaOrType, 2, "string", slashContractAddr)
+	}
+	avsParams.SlashContractAddr = slashContractAddr
+
+	assetID, ok := args[3].(string)
+	if !ok || assetID == "" {
+		return nil, xerrors.Errorf(exocmn.ErrContractInputParaOrType, 3, "string", assetID)
 	}
-	avsParams.OperatorAddress = operatorAddress
-	action, ok := args[3].(uint64)
+	avsParams.AssetID = assetID
+
+	action, ok := args[4].(uint64)
 	if !ok || (action != avstypes.RegisterAction && action != avstypes.DeRegisterAction) {
-		return nil, xerrors.Errorf(exocmn.ErrContractInputParaOrType, 3, "uint64", action)
+		return nil, xerrors.Errorf(exocmn.ErrContractInputParaOrType, 4, "uint64", action)
 	}
 	avsParams.Action = action
 
-	avsOwnerAddress, ok := args[4].(string)
-	if !ok || avsOwnerAddress == "" {
-		return nil, xerrors.Errorf(exocmn.ErrContractInputParaOrType, 4, "string", avsOwnerAddress)
-	}
-	avsOwnerAddress, err = util.ProcessAvsAddress(avsOwnerAddress)
-	if err != nil {
-		return nil, xerrors.Errorf(exocmn.ErrContractInputParaOrType, 1, "[]byte", avsOwnerAddress)
+	minimumDelegation, ok := args[5].(uint64)
+	if !ok || (action != avstypes.RegisterAction && action != avstypes.DeRegisterAction) {
+		return nil, xerrors.Errorf(exocmn.ErrContractInputParaOrType, 5, "uint64", minimumDelegation)
 	}
-	avsParams.AvsOwnerAddress = avsOwnerAddress
+	avsParams.MinimumDelegation = minimumDelegation
 
-	assetID, ok := args[5].(string)
-	if !ok || assetID == "" {
-		return nil, xerrors.Errorf(exocmn.ErrContractInputParaOrType, 3, "uint64", action)
+	unbondingEpochs, ok := args[6].(uint64)
+	if !ok || (action != avstypes.RegisterAction && action != avstypes.DeRegisterAction) {
+		return nil, xerrors.Errorf(exocmn.ErrContractInputParaOrType, 6, "uint64", unbondingEpochs)
 	}
-	avsParams.AssetID = assetID
+	avsParams.UnbondingEpochs = unbondingEpochs
```

Comment:
You defined both `unbondingEpochs` and `Epochs` in protobuf struct, why there's only unbondingEpochs set here?